x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -300,7 +301,7 @@ public static RoleDescriptor parse(String name, BytesReference source, boolean a
 
     public static RoleDescriptor parse(String name, XContentParser parser, boolean allow2xFormat) throws IOException {
         // validate name
-        Validation.Error validationError = Validation.Roles.validateRoleName(name, true);
+        Validation.Error validationError = NativeRealmValidationUtil.validateRoleName(name, true);
```

Comment:
@ywangd I made some changes to address review comments and came across an issue with validation in `RoleDescriptor.parse` method. This method is called from `FileRolesStore, `NativeRolesStore, `ApiKeyService` and `PutRoleRequestBuilder`. This means that using here the "native" validator is wrong.
Since this lowers my confidence in this change, I will mark this PR as draft until I fix this issue and write some more tests for role name validation.

Reply:
@slobodanadamovic This is a good find. I completely missed it when reviewing it previously. My thinking is that we should keep the scope of changes in this PR to the original intention, i.e. fix the role name validation for roles stored in the security index. Based on this, I'd suggest we leave all the current usages of `validateRoleName` and only add the new validation to `PutRoleRequest` and maybe also `NativeRoleStore#putRole`. For any other places, you can come up with a theorectical scenario where it could be a "breaking" change to alter the validation. For example, `ApiKeyService` can take role descriptors with long names because they are not used for document ID, or `TransportPutUserAction` can also take long role names because these roles can from file. Happy to sync if it helps sorting out the issue.

Reply:
> Happy to sync if it helps sorting out the issue.
@ywangd That would help. Let me know what time would be convenient for you.